lib/unfurl-to-attachment.js

```diff
@@ -1,14 +1,39 @@
 const unfurl = require('unfurl.js')
+const UrlParser = require('url')
 
 module.exports = async function unfurlToAttachment (url) {
   const data = await unfurl(url)
+  const parsedUrl = UrlParser.parse(url)
+  let baseUrl, authorIcon
 
-  const description = (data.ogp && data.ogp.ogDescription) || (data.other && data.other.description)
+  if (data.other && data.other.icon) {
+    const parseIconUrl = UrlParser.parse(data.other.icon)
+    if (!parseIconUrl.protocol && !parseIconUrl.host) {
+      baseUrl = parsedUrl.protocol + '//' + parsedUrl.host
+    } else if (!parseIconUrl.protocol && parseIconUrl.host) {
+      baseUrl = parsedUrl.protocol + '//'
+    }
+  }
+  if (baseUrl) {
+    authorIcon = baseUrl + (data.other && data.other.icon)
+  } else {
+    authorIcon = data.other && data.other.icon
+  }
```

Comment:
Looks like this can all be simplified a bit by using the [URL class](https://nodejs.org/docs/latest/api/url.html#url_class_url) in the `url` module:

```
$ node
> const { URL } = require('url')
undefined
> new URL('https://github.com/probot', 'https://example.com').toString()
'https://github.com/probot'
> new URL('/probot', 'https://example.com').toString()
'https://example.com/probot'
```